va/compat_win32.h

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright © Microsoft Corporation
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a
+ * copy of this software and associated documentation files (the "Software"),
+ * to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense,
+ * and/or sell copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice (including the next
+ * paragraph) shall be included in all copies or substantial portions of the
+ * Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL
+ * THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+ * IN THE SOFTWARE.
+ */
+
+#ifndef _COMPAT_WIN32_H_
+#define _COMPAT_WIN32_H_
+
+#include <windows.h>
+#include <winsock.h>
+#include <io.h>
+#include <time.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#define F_OK 0
+
+typedef unsigned int __uid_t;
+
+#ifdef _MSC_VER
+inline char* strtok_r(char *s, const char *delim, char **save_ptr)
+{
+    return strtok_s(s, delim, save_ptr);
+}
+#endif
+
+inline void* dlopen(const char *file, int mode)
+{
+    return LoadLibrary(file);
+}
+
+inline int dlclose (void *handle)
+{
+    return FreeLibrary(handle);
+}
+
+inline void *dlsym (void *handle, const char *name)
+{
+    return GetProcAddress(handle, name);
+}
+
+static char last_err_string[512];

```

Comment:
Scope this within its user. The lifespan will be the same, without making the variable part of the API.